File: javabackend/src/main/java/mft/dev/javabackend/model/Utente.java
package mft.dev.javabackend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Entity
@Table(name = "utenti")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Utente {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(name = "nome")
    private String nome;

    @Column(name = "cognome")
    private String cognome;

    @Column(name = "email", unique = true)
    private String email;

    @OneToMany(mappedBy = "utente", cascade = CascadeType.ALL)
    private List<Todo> todoList;
}

File: kotlinbackend/src/main/kotlin/mft/dev/kotlinbackend/model/Utente.kt
package mft.dev.kotlinbackend.model

import jakarta.persistence.*

@Entity
@Table(name = "utenti")
data class Utente(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
    @Column(name = "nome")
    var nome: String,
    @Column(name = "cognome")
    var cognome: String,
    @Column(name = "email", unique = true)
    var email: String?,
    @OneToMany(mappedBy = "utente", cascade = [CascadeType.ALL])
    var todoList: MutableList<Todo> = mutableListOf()
)
